swarm/api/http/server.go

```diff
@@ -589,6 +643,10 @@ func (s *Server) HandleGetFile(w http.ResponseWriter, r *Request) {
 }
 
 func (s *Server) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	requestCount.Inc(1)
+	startTime := time.Now()
+	defer requestTimer.UpdateSince(startTime)
+	defer metrics.NewResettingTimer("http.request.time_resetting").UpdateSince(startTime)
```

Comment:
Is there a reason to always create a new resetting timer?

Reply:
Also, do we need two timers for the same metric?

Reply:
The reason I left both Timers is to visualise how different the data from both of them would be. This is why we forked the go-metrics library and why we introduced the ResettingTimer.

The default timer will most probably go away in a next PR.

Reply:
We are not creating a new resetting timer always - it is registered once, and then it is retrieved. The `name` `NewResettingTimer` is a bit confusing, but I decided to stick to the naming convention of the `go-ethereum/metrics` package.

I personally think we don't need the variables with counters, gauges, etc. as 99% of them are measured only at a single place in the code, so introducing a variable just serves as a layer of indirection.

Reply:
> The default timer will most probably go away in a next PR.

It would be good to add a TODO as a reminder to remove it (although really I think we should just remove it before merging this PR).

> The name NewResettingTimer is a bit confusing, but I decided to stick to the naming convention of the go-ethereum/metrics package

It is indeed confusing :wink:

> I personally think we don't need the variables with counters, gauges, etc. as 99% of them are measured only at a single place in the code, so introducing a variable just serves as a layer of indirection.

I think having all the metrics declared as variables actually helps in identifying what metrics are being measured. Whatever the choice, let's at least be consistent.

Reply:
Fair enough. I will remove the default timer and declare the resetting timer with the rest of the metrics.

Reply:
Fixed.